juju/juju
# Use names when adding user secrets and include in tabular list

Introduce secret names for user secrets. The name is stored in the label metadata field, but it is exposed as "name" as far as the CLI operates. The update command does not support `--label` any more - it uses `--name` instead to update the secret name.

The add-secret command takes name as the first positional arg. Then, when running update, remove, grant etc, you can use name instead of the URI. If name is used, the URI is retrieved from the name (label).

The tabular list secrets output is also tweaked to show the name for user secrets. And the owner is "<model>" instead of the uuid.

## QA steps

`juju add-secret mysecret foo=bar`

Also add some application and user secrets.

```
$ juju secrets
ID                    Name      Owner         Rotation  Revision  Last updated
ckvrpmip0pbjmiq4jrcg  mysecret  <model>       never            1  22 hours ago  
cl01nsqp0pbiht0p5shg  -         controller    never            1  15 hours ago  
cl0aqtip0pbiht0p5si0  -         controller/0  never            1  5 hours ago  
```

Check that update,remove,grant all work with the name instead of the URI.


## Links

**Jira card:** JUJU-4824
